smac/intensification/intensification.py

```diff
@@ -171,112 +202,133 @@ def eval_challenger(self,
-                                                  incumbent=incumbent,
-                                                  run_history=run_history,
-                                                  log_traj=log_traj)
-
-        except BudgetExhaustedException:
-            # We return incumbent, SMBO stops due to its own budget checks
-            inc_perf = run_history.get_cost(incumbent)
-            self.logger.debug("Budget exhausted; Return incumbent")
-            return incumbent, inc_perf
-
-        # time used to evaluate challenger
-        self.elapsed_time += time.time() - start_time
-
-        # check if 1 intensification run is complete - line 18
-        # this is different to regular SMAC as it requires at least successful challenger run,
-        # which is necessary to work on a fixed grid of configurations.
-        if (
-            self.stage == IntensifierStage.RUN_INCUMBENT
-            and self._chall_indx >= self.min_chall
-            and self.num_chall_run > 0
-        ):
-            if self.num_run > self.run_limit:
-                self.logger.info("Maximum #runs for intensification reached")
-                self._next_iteration()
+        if self.added_incumbent_run:
+            self._ta_time += result.time
+            self.num_run += 1
+            self.added_incumbent_run = False
+            self._process_inc_run(
+                incumbent=incumbent,
+                run_history=run_history,
+                log_traj=log_traj,
+            )
 
-            if not self.use_ta_time_bound and self.elapsed_time - time_bound >= 0:
-                self.logger.info("Wallclock time limit for intensification reached (used: %f sec, available: %f sec)",
-                                 self.elapsed_time, time_bound)
-                self._next_iteration()
+        if self.added_challenger_run:
+            self.num_run += 1
+            self.num_chall_run += 1
+            if result.status == StatusType.CAPPED:
+                # move on to the next iteration
+                self.logger.debug(
+                    "Challenger itensification timed out due "
+                    "to adaptive capping."
+                )
+                self.stage = IntensifierStage.RUN_INCUMBENT
+            else:
 
-            elif self._ta_time - time_bound >= 0:
-                self.logger.info("TA time limit for intensification reached (used: %f sec, available: %f sec)",
-                                 self._ta_time, time_bound)
-                self._next_iteration()
+                self._ta_time += result.time
+                self.added_challenger_run = False
+                incumbent = self._process_racer_results(
+                    challenger=challenger,
+                    incumbent=incumbent,
+                    run_history=run_history,
+                    log_traj=log_traj,
+                )
+
+        elapsed_time = result.endtime - result.starttime
+        if not result.status == StatusType.BUDGETEXHAUSTED:
```

Comment:
Could you use != here? Also, would it be possible to not make the budget exhausted statustype go into the process results function? If the budget is exhausted, nothing should be changed any more, right?

Reply:
Yes, this is here because of the legacy of the old code, but this is something that the smbo can catch and proceed accordingly. I will move this to smbo.